Fixed parameter types

svycka opened the pull request.

svycka committed:
Fixed parameter types

Signed-off-by: Vytautas Stankus <svycka@gmail.com>

michalbundyra commented on src/AbstractValidator.php:
```diff
@@ -321,8 +321,8 @@ protected function createMessage($messageKey, $value)
-     * @param  string $messageKey
-     * @param  string $value      OPTIONAL
+     * @param  string              $messageKey
+     * @param  string|array|object $value      OPTIONAL
```
Also `null` should be noted here, imo.

svycka replied:
then what createMessage() will do as it does not accept null :)

michalbundyra replied:
It's fine, see implementation:

```php
if ($value === null) {
    $value = $this->value;
}
```

svycka committed:
okay forgot null

Signed-off-by: Vytautas Stankus <svycka@gmail.com>

svycka commented:
but default value is still null

On Wed, Sep 30, 2020, 16:14 Michał Bundyra <notifications@github.com> wrote:

> *@michalbundyra* commented on this pull request.
> ------------------------------
>
> In src/AbstractValidator.php
> <https://github.com/laminas/laminas-validator/pull/72#discussion_r497498978>
> :
>
> > @@ -321,8 +321,8 @@ protected function createMessage($messageKey, $value)
>      }
>
>      /**
> -     * @param  string $messageKey
> -     * @param  string $value      OPTIONAL
> +     * @param  string              $messageKey
> +     * @param  string|array|object $value      OPTIONAL
>
> It's fine, see implementation:
>
> if ($value === null) {
>     $value = $this->value;
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/laminas/laminas-validator/pull/72#discussion_r497498978>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJWH7VCYUN3RBDZZ3GM6WDSIMVMLANCNFSM4R7BZPGA>
> .
>

boesing commented:
@svycka Thanks for this PR.

To make travis pass, you can either temporarily add `COMPOSER_ROOT_VERSION=2.14.0` to `travis.yml` or `"version": "2.14.0"` to `composer.json` in a single commit.
After this, travis should startup and execute tests.
When anything is green, we can add this to the next release 👍

svycka commented:
@boesing I am A bit lost do you want something changed? I got an email but I guess you removed the comment.

boesing commented:
Nah, everything is fine. Thats why I deleted my comment.
Just have to sort some things out regarding this project and merging stuff as we have commits in here which should already be part of the branch but isn't...

weierophinney committed:
qa: switch lowest/latest job order for PHP 8 jobs

CONSISTENCY!

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: fix failing tests on "lowest" dependency support

- Updates minimum supported laminas-servicemanager version to 2.7.11
- Do not try and test the BusinessIdentifierCode aliases under v2 releases (untraceable issues when attempting to do so).

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
